app/models/ems_refresh/save_inventory_helper.rb

```diff
@@ -42,6 +42,7 @@ def save_inventory_multi(association, hashes, deletes, find_key, child_keys = []
       deletes.reload
     end
     deletes = deletes.to_a
+    deletes_index = deletes.index_by { |x| x }
```

Comment:
In hindsight, if all you are really doing is trying to ensure efficient lookup, then can you use a `Set`?